src/components/Amalgamation/AmalgamatingBusinesses.vue

```diff
@@ -0,0 +1,255 @@
+          </template>
+          <template v-else>
+            Legal Name: {{ business.name }} <br>
+            Legal Type: {{ business.legalType }} <br>
+            Identifier: {{ business.identifier }} <br>
+            Status: {{ business.status }}
+          </template>
+        </li>
+      </ul>
+    </v-row>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Mixins } from 'vue-property-decorator'
+import { Action, Getter } from 'pinia-class'
+import { useStore } from '@/store/store'
+import { CommonMixin } from '@/mixins'
+import { BusinessLookupServices, LegalServices } from '@/services'
+import { BusinessLookup } from '@bcrs-shared-components/business-lookup'
+import { BusinessIF, BusinessLookupIF, EmptyBusinessLookup } from '@/interfaces'
+
+@Component({
+  components: {
+    BusinessLookup
+  }
+})
+export default class AmalgamatingBusinesses extends Mixins(CommonMixin) {
+  @Getter(useStore) getAmalgamatingBusinesses!: Array<BusinessIF>
+  @Getter(useStore) isAmalgamationFilingHorizontal!: boolean
+  @Getter(useStore) isRoleStaff!: boolean
+
+  @Action(useStore) setAmalgamatingBusinesses!: (x: Array<BusinessIF>) => void
+
+  // Local properties
+  amalgamatingBusinessesValid = false
+  amalgamatingBuinesses = []
+  initialBusinessLookupObject = EmptyBusinessLookup
+
+  // Add an Amalgamating Business button properties
+  isAddingAmalgamatingBusiness = false
+  addAmalgatingBusinessDisabled = false
+
+  // Add an Amalgamating Foreign Business button properties
+  isAddingAmalgamatingForeignBusiness = false
+  addAmalgatingForeignBusinessDisabled = false
+
+  readonly BusinessLookupServices = BusinessLookupServices
+
+  // Cancel button in "Add an Amalgamating Business" is pressed.
+  addAmalgamatingBusinessCancel (): void {
+    this.isAddingAmalgamatingBusiness = false
+    // Enable buttons
+    this.addAmalgatingBusinessDisabled = false
+    this.addAmalgatingForeignBusinessDisabled = false
+  }
+
+  // "Add an Amalgamating Business" button is pressed.
+  onAddBusinessClick (): void {
+    this.isAddingAmalgamatingBusiness = true
+    this.isAddingAmalgamatingForeignBusiness = false
+    // Disable buttons
+    this.addAmalgatingBusinessDisabled = true
+    this.addAmalgatingForeignBusinessDisabled = true
+  }
+
+  // "Add an Amalgamating Foreign Business" button is pressed.
+  onAddForeignBusinessClick (): void {
+    this.isAddingAmalgamatingBusiness = false
+    this.isAddingAmalgamatingForeignBusiness = true
+    // Disable buttons (Please comment out the lines below or remove in part 2)
+    // this.addAmalgatingBusinessDisabled = true
+    // this.addAmalgatingForeignBusinessDisabled = true
+  }
+
+  // Add to the amalgamating businesses array.
+  // If EP (A type), cannot be part of short form horizontal amalgamation.
+  pushToAmalgamatingBusinesses (business: any): void {
+    if (this.isAmalgamationFilingHorizontal) {
+      if (business.legalType !== 'A') this.amalgamatingBuinesses.push(business)
```

Comment:
This is from the note added by Mihai today: "EP (A type) and other Foreign businesses cannot be part of a Short Form Horizontal amalgamation."

This is to fix that.

Reply:
I think we want to add them to the table anyway, where they will fail validation (unless user is staff).